src/Gui/NaviCube.cpp

```diff
@@ -636,20 +647,38 @@ void NaviCubeImplementation::addFace(float gap, const Vector3f& x, const Vector3
 
 	int t = m_VertexArray.size();
 
-	m_VertexArray.emplace_back(z - x - y);
-	m_VertexArray2.emplace_back(z - x2 - y2);
-	m_TextureCoordArray.emplace_back(0, 0);
-	m_VertexArray.emplace_back(z + x - y);
-	m_VertexArray2.emplace_back(z + x2 - y2);
-	m_TextureCoordArray.emplace_back(1, 0);
-	m_VertexArray.emplace_back(z + x + y);
-	m_VertexArray2.emplace_back(z + x2 + y2);
-	m_TextureCoordArray.emplace_back(1, 1);
-	m_VertexArray.emplace_back(z - x + y);
-	m_VertexArray2.emplace_back(z - x2 + y2);
-	m_TextureCoordArray.emplace_back(0, 1);
-
-	// TEX_TOP, TEX_FRONT_FACE, TEX_TOP
+    m_VertexArray.emplace_back(z - x - y);
+    m_TextureCoordArray.emplace_back(0, 0);
+    m_VertexArray.emplace_back(z + x - y);
+    m_TextureCoordArray.emplace_back(1, 0);
+    m_VertexArray.emplace_back(z + x + y);
+    m_TextureCoordArray.emplace_back(1, 1);
+    m_VertexArray.emplace_back(z - x + y);
+    m_TextureCoordArray.emplace_back(0, 1);
+
+    if (pickTex == TEX_FRONT_FACE) {
+        auto x4 = x * (1 - gap * 4);
+        auto y4 = y * (1 - gap * 4);
+        m_VertexArrays2[pickId].emplace_back(z - x2 - y4);
```

Comment:
```suggestion
        auto y4 = y * (1 - gap * 4);
     	m_VertexArrays2[pickId].reserve(8);
        m_VertexArrays2[pickId].emplace_back(z - x2 - y4);
```
Here the needed capacity for the vector is 8. So lets reserve that and prevent a couple of repeated memory allocations and also prevent  overallocation.